texts/0004-wry-raw-window-handle.md

```diff
@@ -0,0 +1,113 @@
+- Feature Name: wry-raw-window-handle
+- Start Date: 16-10-2023
+- RFC PR: [tauri-apps/rfcs#12](https://github.com/tauri-apps/rfcs/pull/12)
+- Tracking Issue: [tauri-apps/wry#677](https://github.com/tauri-apps/wry/issues/677)
+
+# Summary
+
+Make WRY agnostic of the windowing library used and accept a raw-window-handle or gtk-window-handle where applicable.
+
+# Motivation
+
+This has been requested in [tauri-apps/wry#677](https://github.com/tauri-apps/wry/issues/677) a long time ago and at work at CrabNebula,
+we have been contracted to research and develop an integration of wry and wgpu/bevy. We had a successful results and it aligned with [tauri-apps/wry#677](https://github.com/tauri-apps/wry/issues/677)
+so we did an initial implemention of this RFC for Windows which can be found here https://github.com/crabnebula-dev/wry/commits/feat/embedded-webview and we think
+it is a good oprtunity to upstream work.
+
+# Guide-level explanation
+There is two groups of window handles we will need to take in order to support the current platforms we support:
+
+#### Raw window handle:
+This group could leverage the `raw-window-handle` crate; this includes macOS, Windows and Linux (x11) platforms.
+  - Using winit, tao, egui or any other windowing library that
+    implements [`raw_window_handle::HasWindowHandle`](https://docs.rs/raw-window-handle/latest/raw_window_handle/trait.HasWindowHandle.html) trait
+    (or previous versions of `raw-window-handle` crate):
+    ```rs
+    let window = Window::new(&evl);
+    let raw_handle = window.raw_window_handle();
+    let webview = WebView::new(raw_handle);  
```

Comment:
We could take the argument as `impl HasWindowHandle` so it doesn't have to be specific type in raw-window-handle crate

Reply:
I agree, it will make it harder to use with raw win32/x11 handles but it is just one extra step for them but nonetheless it is better 👍 .

Reply:
I added in the rfc a new methods for these raw handles which will be used internally anyways.